### ISSUE TITLE
Question about the common voice results

### ISSUE THREAD

**Issue by wenjie-p** [espnet/espnet#2897]
Hi, I have run the egs/commonvoice/asr1 code using the default config on the German (v5.1-2020-06-22) dataset. The training config is conf/tuning/train_transformer.yaml, I am glad that you have provide the pretrained model and the results. However, I found neither of them is similar to my exp results:

1. The WER of the transformer results on the German test set is 19% as shown in the RESULTS.md, but I got a much lower WER after several(<100) epochs.
2. Then I decide to decode using the pretrained [model](https://drive.google.com/file/d/1IYqKPhaiQ0iy2kPxHej7Q0A3-_DOpC0U/view?usp=sharing), but got a WER of 84.2%.

So my question is:

1. How do you train these models for each language, in the fashion of monolingual or multilingual? FYI, my model was trained in the monolingual fashion.
2. If the model was trained in the monolingual fashion, why the results are so different?

**Comment by ftshijt** [comment]
We already solved the issue via emails. But I would still like to explain the issue here to other curious people.

> How do you train these models for each language, in the fashion of monolingual or multilingual? FYI, my model was trained in the monolingual fashion.

The model published is trained in the monolingual fashion.

> If the model was trained in the monolingual fashion, why the results are so different?
The major difference between RESULTS.md and your experiments is that the result we publish on common-voice did not apply spec-augmentation and speech perturbation. However, in the `run.sh`, we default set those augmentation methods to be on.

> Then I decide to decode using the pretrained model, but got a WER of 84.2%.

We should not observe such large WER, I will go over this model link this week.

**Comment by wenjie-p** [comment]
Hi Jiatong,

My apology for forgetting to mention about this issue in the email. Yes the problem was solved through the email and thanks for the great toolkit you guys have built! If you want any information (e.g. the pretrained models performance details), I would be very happy to help with!

**Comment by wenjie-p** [comment]
Update:

I found that training the `de` German part will cause `nan loss`, during the training of `espnet`, however, the loss is pretty normal. Thus I think `espnet` have done some magic to handle this, after going through the code, I found `espnet` will not update the model if the loss is `nan` (shown in `espnet/espnet/asr/pytorch_backend/asr.py`). The `git log` shows that such functionality was added to `espnet` in Fri Apr 17 22:38:30 2020 with commit id `7cb8a4f4c0dd689ac619d65d86f11acc0f157e0c`.

Thus I suspect that the pretrained model was finished before such commit. I am not sure whether it's the reason but put it here for your reference.

**Comment by wenjie-p** [comment]
Sorry, I found this assumption doesn't hold because common voice `v5.1` was released at 22 June, 2020.

**Comment by ftshijt** [comment]
> Update:
> 
> I found that training the `de` German part will cause `nan loss`, during the training of `espnet`, however, the loss is pretty normal. Thus I think `espnet` have done some magic to handle this, after going through the code, I found `espnet` will not update the model if the loss is `nan` (shown in `espnet/espnet/asr/pytorch_backend/asr.py`). The `git log` shows that such functionality was added to `espnet` in Fri Apr 17 22:38:30 2020 with commit id `7cb8a4f4c0dd689ac619d65d86f11acc0f157e0c`.
> 
> Thus I suspect that the pretrained model was finished before such commit. I am not sure whether it's the reason but put it here for your reference.

Thanks for your comments. Yes, that may not be the problem. As the CTC has a requirement that speech features should be longer than the reference text, the NAN is basically because of failures to meet the requirement.

**Comment by wenjie-p** [comment]
Thanks for the explanation, which makes me much clear because previously I didn't take the CTC loss into account.

Here I also found something about the data split: The downloaded data seems to be accompanied with an official split. After check each split, I found that: train.tsv + dev.tsv + test.tsv << validated.tsv in terms of #utt. As suggested in the [discussion](https://discourse.mozilla.org/t/why-train-tsv-includes-a-few-files-just-3-of-validated-set/36471), the `validated.tsv` contains many repeated sentences (the same transcripts) while [`train.tsv`, `dev.tsv`, `test.tsv`] contains no repeated senteces, the repeated sentences may make the model biased. But there is no clear decision on taking all the repeated data for training or not.

I found `espnet` take all the repeated data in the `validated.tsv` for training (the `dev.tsv` and `test.tsv` excluded), so I am wondering the reason for doing so. Will this make the model biased as suggested in the [discussion](https://discourse.mozilla.org/t/why-train-tsv-includes-a-few-files-just-3-of-validated-set/36471)? Is it an experimental decision or some other consideration?

**Comment by wenjie-p** [comment]
Update:

The final results on my machine trained using `transformer.yaml` with 100 epochs is as below:

```
write a CER (or TER) result in exp/train_de_pytorch_train_transformer_specaug/decode_test_de_decode_lm/result.txt
| SPKR                                           | # Snt # Wrd | Corr    Sub    Del    Ins     Err   S.Err |
| Sum/Avg                                        |15341  544386| 95.1    3.2    1.8    0.8     5.7    44.8 |
write a WER result in exp/train_de_pytorch_train_transformer_specaug/decode_test_de_decode_lm/result.wrd.txt
| SPKR                                           | # Snt # Wrd | Corr    Sub    Del    Ins     Err   S.Err |
| Sum/Avg                                        |15341  137514| 90.2    8.9    1.0    1.0    10.8    44.7 |
```

The above model was trained on the data with kaldi 3-way speed augmentation (default in the `espnet/egs/commonvoice/asr1/run.sh`).

Hope this is helpful for you and others who want to use this recipe.

**Comment by ftshijt** [comment]
> Update:
> 
> The final results on my machine trained using `transformer.yaml` with 100 epochs is as below:
> 
> ```
> write a CER (or TER) result in exp/train_de_pytorch_train_transformer_specaug/decode_test_de_decode_lm/result.txt
> | SPKR                                           | # Snt # Wrd | Corr    Sub    Del    Ins     Err   S.Err |
> | Sum/Avg                                        |15341  544386| 95.1    3.2    1.8    0.8     5.7    44.8 |
> write a WER result in exp/train_de_pytorch_train_transformer_specaug/decode_test_de_decode_lm/result.wrd.txt
> | SPKR                                           | # Snt # Wrd | Corr    Sub    Del    Ins     Err   S.Err |
> | Sum/Avg                                        |15341  137514| 90.2    8.9    1.0    1.0    10.8    44.7 |
> ```
> 
> The above model was trained on the data with kaldi 3-way speed augmentation (default in the `espnet/egs/commonvoice/asr1/run.sh`).
> 
> Hope this is helpful for you and others who want to use this recipe.

That's very cool. Sorry that I did not make the check by the week. Could you set a pull request about your results to the RESULTS.md in the related directory?
